src/hotspot/share/runtime/continuationEntry.cpp

```diff
@@ -36,16 +36,23 @@
 #include "runtime/stubRoutines.hpp"
 
 int ContinuationEntry::_return_pc_offset = 0;
+int ContinuationEntry::_thaw_call_pc_offset = 0;
+int ContinuationEntry::_cleanup_offset = 0;
 address ContinuationEntry::_return_pc = nullptr;
+address ContinuationEntry::_thaw_call_pc = nullptr;
+address ContinuationEntry::_cleanup_pc = nullptr;
 nmethod* ContinuationEntry::_enter_special = nullptr;
 int ContinuationEntry::_interpreted_entry_offset = 0;
 
 void ContinuationEntry::set_enter_code(nmethod* nm, int interpreted_entry_offset) {
   assert(_return_pc_offset != 0, "");
   _return_pc = nm->code_begin() + _return_pc_offset;
+  _thaw_call_pc = nm->code_begin() + _thaw_call_pc_offset;
+  _cleanup_pc = nm->code_begin() + _cleanup_offset;
```

Comment:
I don't see why we need these relative offsets.  Instead of doing
```
_thaw_call_pc_offset = __ pc() - start;
```
why not do
```
_thaw_call_pc = __ pc();
```
The only reason for the offsets would be if what gen_continuation_enter() generated was going to be relocated, but I don't think it is.

Reply:
But these are generated in a temporary buffer. Until we call nmethod::new_native_nmethod() we won't know the final addresses.